src/utils/pgwire/src/pg_response.rs

```diff
@@ -135,6 +143,9 @@ impl StatementType {
                 | StatementType::EXPLAIN
                 | StatementType::SHOW_COMMAND
                 | StatementType::DESCRIBE_TABLE
+                | StatementType::INSERT_RETURNING
+                | StatementType::DELETE_RETURNING
+                | StatementType::UPDATE_RETURNING
```

Comment:
I have a question, as we treat `XXX_RETURNING` as a query. Once they are used in the CTE and the CTE is used multiple times in a single query, can we reuse the CTE result? Reuse means we execute the query once, while not reuse means we execute the query multiple times and then the dml will be executed multiple times.

Reply:
Good catch! So we must "share" this `Insert` and support DAG execution in batch as well? 🥵 Luckily, the parser will reject `RETURNING` in CTE now.

Reply:
Indeed, we need to make sure the dml of the `RETURNING` in CTE will be executed exactly once. That's the PG behavior.